redisvl/utils/vectorize/text/custom.py

```diff
@@ -0,0 +1,293 @@
+import os
+from typing import Any, Callable, Dict, List, Optional
+
+from pydantic.v1 import PrivateAttr
+
+from redisvl.utils.vectorize.base import BaseVectorizer
+
+
+class CustomTextVectorizer(BaseVectorizer):
+    """The CustomTextVectorizer class wraps user-defined embeding methods to create
+    embeddings for text data.
+
+    This vectorizer is designed to accept a provided callable text vectorizer and
+    provides a class definition to allow for compatibility with RedisVL.
+
+    The vectorizer may support both synchronous and asynchronous operations which
+    allows for batch processing of texts, but at a minimum only syncronous embedding
+    is required to satisfy the 'embed()' method.
+
+    .. code-block:: python
+
+        # Synchronous embedding of a single text
+        vectorizer = CustomTextVectorizer(
+            embed = my_vectorizer.generate_embedding
+        )
+        embedding = vectorizer.embed("Hello, world!")
+
+        # Asynchronous batch embedding of multiple texts
+        embeddings = await vectorizer.aembed_many(
+            ["Hello, world!", "How are you?"],
+            batch_size=2
+        )
+
+    """
+
+    _embed_func: Callable = PrivateAttr()
+    _embed_many_func: Optional[Callable] = PrivateAttr()
+    _aembed_func: Optional[Callable] = PrivateAttr()
+    _aembed_many_func: Optional[Callable] = PrivateAttr()
+
+    def __init__(
+        self,
```

Comment:
How would we support `args` and `kwargs` applied to the embedding callables? Could provide these as options here on the init?

Reply:
`args` and `kwargs` should be supported by default if they're defined in the original function. The below snippet works
```
def my_embedder(text:str, some_val:float=42.0):
    return [some_val]
vectorizer = CustomTextVectorizer(embed=my_embedder)
vector = vectorizer.embed('test string', some_val=99.0) # == [99.0]
```
is that your question or am I misunderstanding?